vaadin-master-detail-layout-flow-parent/vaadin-master-detail-layout-flow/src/main/java/com/vaadin/flow/component/masterdetaillayout/MasterDetailLayout.java

```diff
@@ -32,5 +40,169 @@
+        SlotUtils.clearSlot(this, MASTER_SLOT);
+        SlotUtils.addToSlot(this, MASTER_SLOT, component);
+    }
+
+    /**
+     * Gets the component currently in the detail area.
+     *
+     * @return the component in the detail area, or {@code null} if there is no
+     *         component in the detail area
+     */
+    public Component getDetail() {
+        return SlotUtils.getElementsInSlot(this, DETAIL_SLOT).findFirst()
+                .flatMap(Element::getComponent).orElse(null);
+    }
+
+    /**
+     * Sets the component to be displayed in the detail area.
+     *
+     * @param component
+     *            the component to display in the detail area, or {@code null}
+     *            to clear the detail area
+     */
+    public void setDetail(Component component) {
+        doSetDetail(component);
+    }
+
+    private void doSetDetail(HasElement hasElement) {
+        if (hasElement != null && hasElement instanceof Text) {
+            throw new IllegalArgumentException(
+                    "Text as a slot content is not supported. "
+                            + "Consider wrapping the Text inside a Div.");
+        }
+
+        SlotUtils.clearSlot(this, DETAIL_SLOT);
+        if (hasElement != null) {
+            SlotUtils.addToSlot(this, DETAIL_SLOT, hasElement.getElement());
+        }
+    }
+
+    /**
+     * Gets the size of the master area.
+     *
+     * @return the size of the master area in CSS length units, or {@code null}
+     *         if the size is not set
+     */
+    public String getMasterSize() {
+        return getElement().getProperty("masterSize");
+    }
+
+    /**
+     * Sets the size of the master area in CSS length units. When specified, it
+     * prevents the master area from growing or shrinking. If there is not
+     * enough space to show master and detail areas next to each other, the
+     * layout switches to the overlay mode.
+     *
+     * @param size
+     *            the size of the master area in CSS length units
+     */
+    public void setMasterSize(String size) {
+        getElement().setProperty("masterSize", size);
+    }
+
+    /**
+     * Gets the minimum size of the master area.
+     *
+     * @return the minimum size of the master area in CSS length units, or
+     *         {@code null} if the minimum size is not set
+     */
+    public String getMasterMinSize() {
+        return getElement().getProperty("masterMinSize");
+    }
+
+    /**
+     * Sets the minimum size of the master area in CSS length units. When
+     * specified, it prevents the master area from shrinking below this size. If
+     * there is not enough space to show master and detail areas next to each
+     * other, the layout switches to the overlay mode.
+     *
+     * @param minSize
+     *            the minimum size of the master area in CSS length units
+     */
+    public void setMasterMinSize(String minSize) {
```

Comment:
I wonder if we should also provide following alternative:

```java
    public void setMasterMinSize(float minSize, Unit unit) {
```

Reply:
Good point, added overloads for all size setters.